### PR TITLE
Feature- Admin filter photos

### PR DESCRIPTION
**Pull request by GeniJaho**
https://trello.com/c/3sI94kHi
https://trello.com/c/8nQnguJT

- Added filtering photos on Admin Verify Photos page by Country
- Added option to Skip photos
- Removed some informative texts and put them into button tooltips.
- Refactored RecentTags into its own component, and pushed it on the lower left side of the screen.
- When you click on Verify/Delete/False etc on the admin Verify Photos page, and you load the next picture, the page will now scroll on top, not at the bottom of the screen where the user might have scrolled.
- Added 'auth' middleware for all team routes, and fixed a small bug that prevents you from updating your team name/username separately.
- Fixed DeletePhotoAction errors we've received on production. If somehow a photo does not have a filename or a bbox filename, we simply do nothing since the photo doesn't exist. We don't throw an exception since it might have been a partial upload.
- Fixed the Creator of a city/state/country issue showing Anonymous instead of their real name (https://github.com/OpenLitterMap/openlittermap-web/issues/97)

image

### CODE REVIEW

File: app/Http/Controllers/AdminController.php [hunk 1]
```diff
@@ -228,50 +232,98 @@ public function updateTags (Request $request)
...
+        $totalsQuery = Photo::query()
+            ->selectRaw('country_id, count(*) as total')
+            ->whereIn('verification', [0, 0.1])
+            ->whereNotIn('user_id', $this->usersToSkipVerification())
+            ->groupBy('country_id');
+
+        // Using DB to avoid extra appended properties
+        return DB::table('countries')
+            ->selectRaw('id, country, q.total')
+            ->rightJoinSub($totalsQuery, 'q', 'countries.id', '=', 'q.country_id')
```
Comment (xlcrr) [inline app/Http/Controllers/AdminController.php:302]: what is `q` ? query I guess..

File: app/Http/Controllers/PhotosController.php [hunk 2]
```diff
@@ -186,7 +181,7 @@ public function store (Request $request)
         $latitude = $latlong[0];
         $longitude = $latlong[1];
 
-        $revGeoCode = $this->reverseGeocodeAction->run($latitude, $longitude);
+        $revGeoCode = app(ReverseGeocodeLocationAction::class)->run($latitude, $longitude);
```
Comment (xlcrr) [inline app/Http/Controllers/PhotosController.php:184]: any reason why this changed?

File: resources/js/store/modules/admin/actions.js [hunk 3]
```diff
@@ -133,10 +144,27 @@ export const actions = {
...
             .catch(err => {
                 console.error(err);
             });
-    }
+    },
 
+    /**
+     * Get list of countries that contain photos for verification
+     */
+    async GET_COUNTRIES_WITH_PHOTOS (context)
```
Comment (xlcrr) [inline resources/js/store/modules/admin/actions.js:158]: I think all admin action names should be prefixed with `ADMIN_` to make it clear this is not a normal user operation

**Review by xlcrr — APPROVED** [review]

File: app/Http/Controllers/AdminController.php (hunk 1 above)
Comment (GeniJaho) [reply app/Http/Controllers/AdminController.php:302]: Yes, every subquery needs an alias, 'q' is quite common.

File: app/Http/Controllers/PhotosController.php (hunk 2 above)
Comment (GeniJaho) [reply app/Http/Controllers/PhotosController.php:184]: Yes, I added a fake for this action, which allows me more options to properly mock the geocoding action. I don't need to mock this class in every test anymore. Using dependency injection without an interface forces me to use app() in this case. Nothing wrong with it though, this code piece is perfectly fine.

**Pull request merged**